### PR TITLE
feat: implementing SubmitTx gRPC endpoint

### PR DESCRIPTION
**Pull request by seantking**
## Description

closes: #2056 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by seantking** [comment]
Shall I add an e2e for this test in a follow up PR?

**Comment by colin-axner** [comment]
> Shall I add an e2e for this test in a follow up PR?

#2039, #2063

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server.go [hunk 1]
```diff
@@ -26,5 +30,27 @@ func (k Keeper) RegisterAccount(goCtx context.Context, msg *types.MsgRegisterAcc
...
+	if !found {
+		return nil, sdkerrors.Wrapf(icatypes.ErrActiveChannelNotFound, "failed to retrieve active channel for port %s", portID)
+	}
+
+	chanCap, found := k.scopedKeeper.GetCapability(ctx, host.ChannelCapabilityPath(portID, channelID))
+	if !found {
+		return nil, sdkerrors.Wrap(channeltypes.ErrChannelCapabilityNotFound, "module does not own channel capability")
+	}
+
+	_, err = k.SendTx(ctx, chanCap, msg.ConnectionId, portID, msg.PacketData, msg.TimeoutTimestamp)
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/keeper/msg_server.go:50]: We can include the sequence in the `MsgSubmitTxResponse`, right?

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 
-	icatypes "github.com/cosmos/ibc-go/v5/modules/apps/27-interchain-accounts/controller/types"
+	"github.com/cosmos/ibc-go/v5/modules/apps/27-interchain-accounts/controller/types"
+	icacontrollertypes "github.com/cosmos/ibc-go/v5/modules/apps/27-interchain-accounts/controller/types"
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go:11]: nit: I think we normally alias this as just `controllertypes`

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go [hunk 3]
```diff
@@ -85,3 +92,108 @@ func (suite *KeeperTestSuite) TestRegisterAccount() {
 		}
 	}
 }
+
+func (suite *KeeperTestSuite) TestSubmitTx() {
+	var (
+		path         *ibctesting.Path
+		owner        string
+		connectionID string
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go:100]: do we need to expose this var to mutate, or can we just use `path.EndpointA.ConnectionID` and mutate the path in `malleate()`?

**Review by damiannolan — COMMENTED** [review]
Looks more or less good to go, left some nits on tests and we should return sequence in the msg response afaik!
Nice work!

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go (hunk 3 above)
Comment (seantking) [reply modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go:100]: I think it's fine to leave as is :handshake:

**Review by damiannolan — APPROVED** [review]

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go (hunk 3 above)
Comment (seantking) [reply modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go]: You're right actually, I updated it. Thanks

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server.go [hunk 4]
```diff
@@ -26,5 +30,27 @@ func (k Keeper) RegisterAccount(goCtx context.Context, msg *types.MsgRegisterAcc
...
+
+	portID, err := icatypes.NewControllerPortID(msg.Owner)
+	if err != nil {
+		return nil, err
+	}
+
+	channelID, found := k.GetActiveChannelID(ctx, msg.ConnectionId, portID)
+	if !found {
+		return nil, sdkerrors.Wrapf(icatypes.ErrActiveChannelNotFound, "failed to retrieve active channel for port %s", portID)
+	}
```
[the remark this answers is not in the document]
Comment (AdityaSripal) [reply modules/apps/27-interchain-accounts/controller/keeper/msg_server.go:48]: I suppose we could, but i think in that future we will break APIs anyway. So not sure its worth optimizing for that now

**Pull request merged**